### PR TITLE
Remove deprecated traits ending with version 25.01 (#346)

### PR DESCRIPTION
**Pull request by jtodev**
### Description

Traits were removed from:
* Grid
* RectGrid3D
* PowerSpectra
* PointSource
* UncorrelatedNoiseSource
* SineGenerator
* BeamformerTime
* BeamformerBase
* PointSpreadFunction

### Reference issue

#346 
### Checklist

- [x] I have read the [Contributing](https://www.acoular.org/contributing/index.html) section.
- [x] My branch is up-to-date with the *master* branch of the [Acoular repository](https://github.com/acoular/acoular).
- [x] My changes fulfill the [Code Quality](https://www.acoular.org/contributing/quality.html) standards.
- [x] I have updated the [What's new](https://www.acoular.org/news/index.html) section the the [documentation](https://github.com/acoular/acoular/blob/master/docs/source/news/index.rst) explaining my changes.
- [x] I have appended myself to the [CITATION.cff](https://github.com/acoular/acoular/blob/master/CITATION.cff) file.

### CODE REVIEW

File: acoular/fbeamform.py [hunk 1]
```diff
@@ -310,126 +310,7 @@ class BeamformerBase(HasPrivateTraits):
...
-    def _get_rm(self):
-        msg = (
-            "Deprecated use of 'rm' trait. Please use the 'steer' trait with an object of class"
-            "'SteeringVector'. The 'rm' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        return self._steer_obj.rm
-
-    # --- End of backwards compatibility traits --------------------------------------
+    steer = Instance(SteeringVector(), SteeringVector)
```
Comment (adku1173) [inline acoular/fbeamform.py:313]: Please use this expression. (Steering vector instance is created when BeamformerBase instance is created)

```suggestion
    steer = Instance(SteeringVector, args=())
```

File: acoular/fbeamform.py [hunk 2]
```diff
@@ -1083,131 +964,14 @@ class PointSpreadFunction(HasPrivateTraits):
...
```
Comment (adku1173) [inline acoular/fbeamform.py:967]:
```suggestion
    steer = Instance(SteeringVector, args=())
```

File: acoular/spectra.py [hunk 3]
```diff
@@ -176,12 +176,7 @@ class PowerSpectra(BaseSpectra):
 
     #: The :class:`~acoular.calib.Calib` object that provides the calibration data,
     #: defaults to no calibration, i.e. the raw time data is used.
-    #: **deprecated, will be removed in version 25.01**: use
-    #: :attr:`~acoular.sources.TimeSamples.calib` property of :class:`~acoular.sources.TimeSamples`
-    #: objects
-    calib = Property(desc='calibration object (deprecated, will be removed in version 25.01)')
-
-    _calib = Instance(Calib)
+    calib = Instance(Calib)
```
Comment (adku1173) [inline acoular/spectra.py:179]: calib capabilities should be removed from PowerSpectra (requires changes in `calc_csm` method)

File: acoular/spectra.py [hunk 4]
```diff
@@ -316,20 +299,9 @@ def _set_ind_low(self, ind_low):  # by setting this the user sets the higher ind
         self._index_set_last = True
         self._ind_low = ind_low
 
-    def _set_time_data(self, time_data):
```
Comment (adku1173) [inline acoular/spectra.py:319]: Did you remove the time_data attribute as well?

File: acoular/tbeamform.py [hunk 5]
```diff
@@ -102,141 +100,23 @@ class BeamformerTime(TimeOut):
...
-        msg = (
-            "Deprecated use of 'rm' trait. Please use the 'steer' trait with an object of class  "
-            "'SteeringVector'. The 'rm' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        return self._steer_obj.rm
-
-    # --- End of backwards compatibility traits --------------------------------------
-
+    steer = Instance(SteeringVector(), SteeringVector)
```
Comment (adku1173) [inline acoular/tbeamform.py:103]:
```suggestion
    steer = Instance(SteeringVector, args=())
```

File: acoular/tbeamform.py [hunk 6]
```diff
@@ -102,141 +100,23 @@ class BeamformerTime(TimeOut):
...
     #: Number of channels in output (=number of grid points).
-    numchannels = Delegate('grid', 'size')
+    numchannels = Property(depends_on=['steer.grid'])
```
Comment (adku1173) [inline acoular/tbeamform.py:105]: This line does not work as expected. Numchannels only changes when the steer.grid class changes, but not if the grid size changes.

**Review by adku1173 — CHANGES_REQUESTED** [review]

**Comment by adku1173** [comment]
@jtodev rebase required

**Comment by adku1173** [comment]
@jtodev rebase required

File: acoular/spectra.py [hunk 7]
```diff
@@ -179,12 +179,7 @@ class PowerSpectra(BaseSpectra):
 
     #: The :class:`~acoular.calib.Calib` object that provides the calibration data,
     #: defaults to no calibration, i.e. the raw time data is used.
-    #: **deprecated, will be removed in version 25.01**: use
-    #: :attr:`~acoular.sources.TimeSamples.calib` property of :class:`~acoular.sources.TimeSamples`
-    #: objects
-    calib = Property(desc='calibration object (deprecated, will be removed in version 25.01)')
-
-    _calib = Instance(Calib)
+    calib = Instance(Calib)
```
Comment (adku1173) [inline acoular/spectra.py:182]: Please remove the calib capabilities from the PowerSpectra class (you can parse for calib):
* remove the calib attribute
* remove the _handle_dual_calibration, _get_calib and _set_calib functions
* remove calib attributes from all sublcasses (including PowerSpectraImport)

**Review by adku1173 — CHANGES_REQUESTED** [review]

File: acoular/tbeamform.py (hunk 6 above)
Comment (adku1173) [reply acoular/tbeamform.py]: This line is still included.

File: acoular/spectra.py (hunk 4 above)
Comment (adku1173) [reply acoular/spectra.py]: time_data attribute is still part of the PowerSpectra class and its subclasses

**Comment by adku1173** [comment]
@artpelling ready to be reviewed / merged

File: acoular/fbeamform.py (hunk 2 above)
Comment (artpelling) [reply acoular/fbeamform.py:974]: @jtodev Please change this as @adku1173 suggested, it is the recommended way. See https://docs.enthought.com/traits/traits_user_manual/defining.html#instance

File: acoular/tbeamform.py [hunk 8]
```diff
@@ -102,141 +100,23 @@ class BeamformerTime(TimeOut):
...
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        return self._steer_obj.rm
-
-    # --- End of backwards compatibility traits --------------------------------------
+    steer = Instance(SteeringVector, args=())
 
     #: Number of channels in output (=number of grid points).
-    num_channels = Delegate('grid', 'size')
+    num_channels = Property()
```
Comment (artpelling) [inline acoular/tbeamform.py:106]: @adku1173  @jtodev  Is this correct? Why not use `Delegate` and if not doesnt this need a `depends_on`?

**Review by artpelling — CHANGES_REQUESTED** [review]

File: acoular/tbeamform.py (hunk 8 above)
Comment (adku1173) [reply acoular/tbeamform.py:106]: This is correct ;). depends on is not needed because nothing is cached.

File: acoular/fbeamform.py (hunk 2 above)
Comment (adku1173) [reply acoular/fbeamform.py]: Done

**Comment by adku1173** [comment]
@artpelling should be ready now.

**Review by adku1173 — APPROVED** [review]

**Pull request merged**